### ISSUE TITLE
SPIKE: Explore APIs for LMS [5,5,1]

### ISSUE THREAD

**Issue by rsgonzal** [Iridescent-CM/technovation-app#2878]
Our main goal and most important need is that we have a single sign on option
- We can create their accounts with the same login as on the TG platform
- They can be automatically signed on everytime they click a link from our site to theirs

Nice to haves
- Theming on the LMS site, looks just like ours (with new branding)
- We can pass information to the site on teams
- We can pull things from the LMS site (like pieces of submission)

[Doc for LMS exploration
](https://docs.google.com/document/d/1FoAtT0bpTNnxkeNIW12pyiLpXP2pYq5FD9Lax35oXug/edit?usp=sharing)
@nickiredescent @hellafitz Does this summarize what we've been exploring?

**Comment by shaunxp20** [comment]
> * We can create their accounts with the same login as on the TG platform
> * They can be automatically signed on everytime they click a link from our site to theirs

It looks like both LMSs we're looking at support this criteria.

From LearnWorld's documentation on [SSO](https://docs.learnworlds.com/#sso-with-a-user-39-s-email-or-user-id):
> If a user does not exist then one will be automatically be created...

From Thinkific's documentation on [SSO](https://developers.thinkific.com/more/sso/#how-it-works): 
> ...if they don’t already have an account on your Thinkific site, creates an account and signs them in.

So we should be able to create an account for them.

-----

> Nice to haves
> * Theming on the LMS site, looks just like ours (with new branding)

Both LMSs have theming capabilities.

> * We can pass information to the site on teams
> * We can pull things from the LMS site (like pieces of submission)

Both have APIs, LearnWorlds is rather limited, just around [courses](https://docs.learnworlds.com/#courses) and [users](https://docs.learnworlds.com/#users) (creating, getting, updating enrolling, un-enrolling, suspending, un-suspending). Thinkific's API seems more robust comparatively: https://developers.thinkific.com/api/api-documentation/

At this point, they both look good (at least in regards to SSO and theming), I'm thinking it would make more sense to explore/spike more after we've picked one of them. 😄

**Comment by rsgonzal** [comment]
Vivian added notes in google doc

**Comment by rsgonzal** [comment]
Brightspace
- does have the SSo capabilities
- documentation not very robust, hard to follow, old
- not as impressive APIs
- can not customize as much as the other two
- good apis, can do a lot, but figuring it out is going to be difficult
